lib/services/local/externalauditstorage.go

```diff
@@ -109,22 +113,28 @@ func (s *ExternalAuditStorageService) UpsertDraftExternalAuditStorage(ctx contex
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	item := backend.Item{
+		Key:   draftExternalAuditStorageBackendKey,
+		Value: value,
+	}
 
-	// Lock is used here and in Promote to prevent upserting in the middle
-	// of a promotion and the possibility of deleting a newly upserted draft
-	// after the previous one was promoted.
-	err = backend.RunWhileLocked(ctx, backend.RunWhileLockedConfig{
-		LockConfiguration: backend.LockConfiguration{
-			Backend:  s.backend,
-			LockName: externalAuditStorageLockName,
-			TTL:      externalAuditStorageLockTTL,
+	integrationKey, integrationRevision, err := s.checkAWSIntegration(ctx, in.Spec.IntegrationName)
+	if err != nil {
+		return nil, trace.Wrap(err, "checking AWS OIDC integration")
+	}
+	_, err = s.backend.AtomicWrite(ctx, []backend.ConditionalAction{
```

Comment:
If the `AtomicWrite` fails due to a concurrent change of the relevant OIDC integration we should probably try again, and only fail after some iterations (3? 5?), or if `checkAWSIntegration` detects an invalid state; same for `CreateDraftExternalAuditStorage`.

Reply:
These things change so infrequently and slowly, is it really necessary? And you probably really don't want to be editing your EAS configuration live while someone else is concurrently editing the OIDC integrations

Reply:
It's just a little weird to get `CompareFailed` from a `Delete` or `Upsert` operation, but EAS config is something that's going to be written exactly zero times or one time for basically all cloud tenants anyway, so we can just decide that it's ok to get `CompareFailed`.